spring-xd-dirt/src/main/java/org/springframework/xd/dirt/stream/XDStreamParser.java

```diff
@@ -322,4 +333,98 @@ private ModuleType resolveModuleType(String moduleName, ModuleType... candidates
 		throw new NoSuchModuleException(moduleName, candidates);
 	}
 
+	/**
+	 * A wrapper around the canonical parser that supports multiline "document" parsing.
+	 *
+	 * <p>While XDStreamParser parses a single definition, this class is able to parse multiple, reporting
+	 * errors or accepting constructs like taps against definitions that appear earlier in the document (without
+	 * actually saving them in the main repository).</p>
+	 *
+	 * @author Eric Bottard
+	 */
+	public static class MultiLineDocumentParser {
+
+		private final XDStreamParser delegate;
+
+		public MultiLineDocumentParser(XDStreamParser delegate) {
+			this.delegate = delegate;
+		}
+
+		public DocumentParseResult parse(String[] document) {
+			DocumentParseResult result = new DocumentParseResult(document.length);
+			CrudRepository<BaseDefinition, String> transientRepository = new TransientDefinitionRepository();
+			StreamConfigParser parser = new StreamConfigParser(transientRepository);
+			for (String nameAndDefinitionPair : document) {
+				try {
+
+					StreamNode stream = parser.parse(nameAndDefinitionPair);
```

Comment:
ok. I have created https://jira.spring.io/browse/XD-3059 to track the support to parse multi line content with both stream and job definitions.